src/Tracking/TrackingData/DonationFormsData.php

```diff
@@ -100,66 +113,96 @@ public function setFormIdsByDonationIds() {
 				AND donation_id IN ({$donationIdsList})
 				"
 		);
+
+		return $this;
 	}
 
 	/**
-	 * Returns revenue till current date.
+	 * Set forms revenues.
 	 *
-	 * @param int $formId
-	 *
-	 * @return string
 	 * @since 2.10.0
+	 * @return self
 	 */
-	public function getRevenueTillNow( $formId ) {
+	private function setRevenues() {
 		global $wpdb;
 
-		$statues = ArrayDataSet::getStringSeparatedByCommaEnclosedWithSingleQuote(
-			[
-				'publish', // One time donation
-				'give_subscription', // Renewal
-			]
+		$formIds       = ArrayDataSet::getStringSeparatedByCommaEnclosedWithSingleQuote( $this->formIds );
+		$defaultResult = array_combine(
+			$this->formIds,
+			array_fill( 0, count( $this->formIds ), 0 ) // Set default revenue to 0
 		);
 
-		$result = (int) $wpdb->get_var(
-			$wpdb->prepare(
-				"
-				SELECT SUM(amount)
-				FROM {$wpdb->give_revenue} as r
-				INNER JOIN {$wpdb->posts} as p
-				ON r.donation_id=p.id
-				INNER JOIN {$wpdb->donationmeta} as dm
-				ON p.id=dm.donation_id
-				AND post_status IN ({$statues})
-				AND r.form_id=%d
-				AND dm.meta_key='_give_payment_mode'
-				AND dm.meta_value='live'
-				",
-				$formId
-			)
+		$result = $wpdb->get_results(
+			"
+			SELECT SUM(amount) as amount, form_id
+			FROM {$wpdb->give_revenue}
+			WHERE form_id IN ({$formIds})
+			GROUP BY form_id
+			",
+			ARRAY_A
 		);
-		return $result ?: 0;
+
+		if ( $result ) {
+			$result = array_map(
+				'absint',
+				array_combine(
+					wp_list_pluck( $result, 'form_id' ),
+					wp_list_pluck( $result, 'amount' )
+				)
+			);
+		}
+
+		$this->formRevenues = array_replace( $defaultResult, $result );
+
+		return $this;
 	}
 
 	/**
-	 * Get donor count.
+	 * Set forms revenues till current date.
 	 *
 	 * @since 2.10.0
 	 *
-	 * @param int $formId
-	 *
-	 * @return int
+	 * @return self
 	 */
-	private function getDonorCount( $formId ) {
-		$donorQuery = new Give_Donors_Query(
-			[
-				'number'          => -1,
-				'count'           => true,
-				'donation_amount' => 0,
-				'give_forms'      => [ $formId ],
-			]
+	private function setDonorCounts() {
+		global $wpdb;
+
+		$formIds       = ArrayDataSet::getStringSeparatedByCommaEnclosedWithSingleQuote( $this->formIds );
+		$defaultResult = array_combine(
+			$this->formIds,
+			array_fill( 0, count( $this->formIds ), 0 ) // Set default donor count to 0
+		);
+
+		$result = $wpdb->get_results(
+			"
+			SELECT COUNT(DISTINCT dm2.meta_value) as donor_count, dm.meta_value as form_id
+			FROM {$wpdb->donationmeta} as dm
+			INNER JOIN {$wpdb->donationmeta} as dm2
+			ON dm.donation_id = dm2.donation_id
+			INNER JOIN {$wpdb->donors} as donor
+			ON dm2.meta_value = donor.id
+			WHERE dm.meta_key='_give_payment_form_id'
+			AND dm.meta_value IN ({$formIds})
+			AND dm2.meta_key='_give_payment_donor_id'
+			AND donor.purchase_value > 0
+			GROUP BY dm.meta_value
```

Comment:
Indents can go a long way in making SQL more readable:
```suggestion
			SELECT COUNT(DISTINCT dm2.meta_value) as donor_count, dm.meta_value as form_id
			FROM {$wpdb->donationmeta} as dm
				INNER JOIN {$wpdb->donationmeta} as dm2 ON dm.donation_id = dm2.donation_id
				INNER JOIN {$wpdb->donors} as donor ON dm2.meta_value = donor.id
			WHERE dm.meta_key='_give_payment_form_id'
				AND dm.meta_value IN ({$formIds})
				AND dm2.meta_key='_give_payment_donor_id'
				AND donor.purchase_value > 0
			GROUP BY dm.meta_value
```